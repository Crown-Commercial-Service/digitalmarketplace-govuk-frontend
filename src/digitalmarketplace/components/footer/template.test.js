/**
 * @jest-environment jsdom
 */
const axe = require('../../../../lib/axe-helper')

const { render, getExamples } = require('../../../../lib/jest-helpers.js')

const examples = getExamples('footer')

const urlForMock = jest.fn((param) => {
  let value = ''
  if (param === 'external.help') {
    value = 'http://www.example.com/1'
  } else if (param === 'external.suppliers_list_by_prefix') {
    value = 'http://www.example.com/2'
  } else if (param === 'external.cookies') {
    value = 'http://www.example.com/3'
  } else if (param === 'external.privacy_notice') {
    value = 'http://www.example.com/4'
  } else if (param === 'external.terms_and_conditions') {
    value = 'http://www.example.com/5'
  } else if (param === 'external.accessibility_statement') {
    value = 'http://www.example.com/6'
  }

  return value
})

describe('footer', () => {
  it('default example passes accessibility tests', async () => {
    const $ = render('footer', examples.default, { url_for: urlForMock })

    const results = await axe($.html())
    expect(results).toHaveNoViolations()
  })

  it('renders a footer component with all our standard links', () => {
    const $ = render('footer', examples.default, { url_for: urlForMock })
    expect($.html()).toMatchSnapshot()
  })
})
