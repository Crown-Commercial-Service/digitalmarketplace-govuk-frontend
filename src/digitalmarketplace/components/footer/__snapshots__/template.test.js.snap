// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer renders a footer component with all our standard links 1`] = `
"<html><head></head><body><footer class=\\"govuk-footer \\" role=\\"contentinfo\\">
  <div class=\\"govuk-width-container \\">
      <div class=\\"govuk-footer__navigation\\">
          <div class=\\"govuk-footer__section\\">
            <h2 class=\\"govuk-footer__heading govuk-heading-m\\">About Digital Marketplace</h2>
              <ul class=\\"govuk-footer__list \\">
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"http://www.google.com\\">
                        Contact
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/guidance/digital-marketplace-buyers-guide\\">
                        Services you can buy
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/guidance/digital-marketplace-suppliers-guide\\">
                        Services you can sell
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/government/organisations/crown-commercial-service\\">
                        About Crown Commercial Services
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/government/organisations/government-digital-service\\">
                        About Government Digital Services
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"http://www.gov.uk\\">
                        G-Cloud supplier A to Z
                      </a>
                    </li>
              </ul>
          </div>
          <div class=\\"govuk-footer__section\\">
            <h2 class=\\"govuk-footer__heading govuk-heading-m\\">Guidance for using Digital Marketplace</h2>
              <ul class=\\"govuk-footer__list \\">
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/guidance/g-cloud-suppliers-guide\\">
                        Applying to sell on the G-Cloud framework
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/guidance/digital-outcomes-and-specialists-suppliers-guide\\">
                        Applying to sell on the DOS framework
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/guidance/how-to-sell-your-digital-outcomes-and-specialists-services\\">
                        Responding to buyer requirements on the DOS framework
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/guidance/g-cloud-buyers-guide\\">
                        Buying on the G-Cloud framework
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/guidance/digital-outcomes-and-specialists-buyers-guide\\">
                        Buying on the DOS framework
                      </a>
                    </li>
                    <li class=\\"govuk-footer__list-item\\">
                      <a class=\\"govuk-footer__link\\" href=\\"https://www.gov.uk/guidance/the-crown-hosting-data-centres-framework-on-the-digital-marketplace\\">
                        Buying on the Crown Hosting framework
                      </a>
                    </li>
              </ul>
          </div>
      </div>
      <hr class=\\"govuk-footer__section-break\\">
    <div class=\\"govuk-footer__meta\\">
      <div class=\\"govuk-footer__meta-item govuk-footer__meta-item--grow\\">
          <h2 class=\\"govuk-visually-hidden\\">Support links</h2>
            <ul class=\\"govuk-footer__inline-list\\">
                <li class=\\"govuk-footer__inline-list-item\\">
                  <a class=\\"govuk-footer__link\\" href=\\"http://www.yahoo.co.uk\\">
                    Terms and conditions
                  </a>
                </li>
                <li class=\\"govuk-footer__inline-list-item\\">
                  <a class=\\"govuk-footer__link\\" href=\\"http://www.bbc.co.uk\\">
                    Cookies
                  </a>
                </li>
                <li class=\\"govuk-footer__inline-list-item\\">
                  <a class=\\"govuk-footer__link\\" href=\\"http://www.amazon.co.uk\\">
                    Privacy notice
                  </a>
                </li>
            </ul>
            <div class=\\"govuk-footer__meta-custom\\">
              Built by the <a href=\\"https://www.gov.uk/government/organisations/government-digital-service\\" class=\\"govuk-footer__link\\">Government Digital Service</a>
            </div>

        <svg aria-hidden=\\"true\\" focusable=\\"false\\" class=\\"govuk-footer__licence-logo\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 483.2 195.7\\" height=\\"17\\" width=\\"41\\">
          <path fill=\\"currentColor\\" d=\\"M421.5 142.8V.1l-50.7 32.3v161.1h112.4v-50.7zm-122.3-9.6A47.12 47.12 0 0 1 221 97.8c0-26 21.1-47.1 47.1-47.1 16.7 0 31.4 8.7 39.7 21.8l42.7-27.2A97.63 97.63 0 0 0 268.1 0c-36.5 0-68.3 20.1-85.1 49.7A98 98 0 0 0 97.8 0C43.9 0 0 43.9 0 97.8s43.9 97.8 97.8 97.8c36.5 0 68.3-20.1 85.1-49.7a97.76 97.76 0 0 0 149.6 25.4l19.4 22.2h3v-87.8h-80l24.3 27.5zM97.8 145c-26 0-47.1-21.1-47.1-47.1s21.1-47.1 47.1-47.1 47.2 21 47.2 47S123.8 145 97.8 145\\"/>
        </svg>
        <span class=\\"govuk-footer__licence-description\\">
          All content is available under the
          <a class=\\"govuk-footer__link\\" href=\\"https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/\\" rel=\\"license\\">Open Government Licence v3.0</a>, except where otherwise stated
        </span>
      </div>
      <div class=\\"govuk-footer__meta-item\\">
        <a class=\\"govuk-footer__link govuk-footer__copyright-logo\\" href=\\"https://www.nationalarchives.gov.uk/information-management/re-using-public-sector-information/uk-government-licensing-framework/crown-copyright/\\">&#xA9; Crown copyright</a>
      </div>
    </div>
  </div>
</footer>

</body></html>"
`;
